Tests/CoordinateTransformationsTest/CoordinateTransformationsTest.cpp

```diff
@@ -0,0 +1,180 @@
+    Tensor<2, double> Mij_cart;
+    FOR2(i, j) { Mij_cart[i][j] = 0.; }
+    Mij_cart[0][0] = 1.;
+    Mij_cart[1][1] = 1.;
+    Mij_cart[2][2] = 1.;
+
+    Tensor<2, double> Mij_spher;
+    FOR2(i, j) { Mij_spher[i][j] = 0.; }
+    Mij_spher[0][0] = 1.;
+    Mij_spher[1][1] = r * r;
+    Mij_spher[2][2] = r2sin2theta;
+
+    // Test cartesian_to_spherical_LL
+    Tensor<2, double> Mij_spher_check;
+    Mij_spher_check = cartesian_to_spherical_LL(Mij_cart, x, y, z);
+    failed =
+        check_tensor(Mij_spher_check, Mij_spher, "cartesian_to_spherical_LL");
+
+    // Test spherical_to_cartesian_LL
+    Tensor<2, double> Mij_cart_check;
+    Mij_cart_check = spherical_to_cartesian_LL(Mij_spher, x, y, z);
+    failed =
+        check_tensor(Mij_cart_check, Mij_cart, "spherical_to_cartesian_LL");
+
+    // Test cartesian_to_spherical_UU
+    Tensor<2, double> Mij_spher_UU;
+    Tensor<2, double> Mij_spher_UU_check;
+    Mij_spher_UU_check =
+        cartesian_to_spherical_UU(compute_inverse_sym(Mij_cart), x, y, z);
+    Mij_spher_UU = compute_inverse_sym(Mij_spher);
+    failed = check_tensor(Mij_spher_UU_check, Mij_spher_UU,
+                          "cartesian_to_spherical_UU");
+
+    // Test spherical_to_cartesian_UU
+    Tensor<2, double> Mij_cart_UU;
+    Tensor<2, double> Mij_cart_UU_check;
+    Mij_cart_UU_check =
+        spherical_to_cartesian_UU(compute_inverse_sym(Mij_spher), x, y, z);
+    Mij_cart_UU = compute_inverse_sym(Mij_cart);
+    failed = check_tensor(Mij_cart_UU_check, Mij_cart_UU,
+                          "spherical_to_cartesian_UU");
+
+    // Test vector transformations
+    Tensor<1, double> si_cart_U;
+    si_cart_U[0] = x / r;
+    si_cart_U[1] = y / r;
+    si_cart_U[2] = z / r;
+
+    Tensor<1, double> si_spher_U;
+    si_spher_U[0] = 1.0;
+    si_spher_U[1] = 0.0;
+    si_spher_U[2] = 0.0;
+
+    // Test cartesian_to_spherical_U
+    Tensor<1, double> si_spher_U_check;
+    si_spher_U_check = cartesian_to_spherical_U(si_cart_U, x, y, z);
+    failed =
+        check_vector(si_spher_U_check, si_spher_U, "cartesian_to_spherical_U");
+
+    // Test spherical_to_cartesian_U
+    Tensor<1, double> si_cart_U_check;
+    si_cart_U_check = spherical_to_cartesian_L(si_spher_U, x, y, z);
+    failed =
+        check_vector(si_cart_U_check, si_cart_U, "spherical_to_cartesian_U");
+
+    // Test area_element_sphere
+    double area_element = r * sqrt(rho2);
+    double area_element_check = area_element_sphere(Mij_spher);
+    if (!almost_equal(area_element, area_element_check, ulp))
+    {
+        std::cout << "Failed area_element\n";
+        std::cout << "value: " << area_element << "\n";
+        std::cout << "correct_value: " << area_element_check << std::endl;
+        failed = -1;
+    }
+
+    if (failed == 0)
+        std::cout << "Coordinate transformations test passed..." << std::endl;
+    else
+        std::cout << "Coordinate transformations test failed..." << std::endl;
+
+    return failed;
```

Comment:
One should be careful with the `failed`. If I'm not wrong, if any of the first tests fail, this will be overridden by later tests.
Idea: convert `failed` to a bool and then do `failed |= check...(...)`. Or whatever else

Reply:
Ah right, that was the case, should work now, thanks for the suggestion :)